### PR TITLE
Add host list, add/edit forms, and details

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
Host List:
Screen Shot 2019-10-30 at 4 15 02 PM

Host Details:
Screen Shot 2019-10-30 at 4 15 22 PM

Host Add:
Screen Shot 2019-10-30 at 4 15 41 PM

Host Edit:
Screen Shot 2019-10-30 at 4 16 04 PM

Note that the toggles on the host list are currently borked after the recent pf upgrade.  @keithjgrant is taking a look at that and will be addressed in a separate PR.

@ryanpetrello there's an api change in here to add `type` to the recent_jobs for hosts returned in `/api/v2/hosts`.  I admit that I don't really know what I'm doing when it comes to python so I'd appreciate a review from someone on your team to make sure I'm not doing something silly.

##### ISSUE TYPE
 - Feature Pull Request

### CODE REVIEW

**Comment by keithjgrant** [comment]
Small bug: save host variables as JSON, then click edit -> the form displays JSON, but the toggle says YAML (toggling back & forth still works and will correct the discrepancy):

Screen Shot 2019-11-01 at 10 17 15 AM

...this is probably a bug in the VariablesField detecting they data type on load. Should I tackle this in a separate issue or do you want to look at it here?

File: awx/ui_next/src/screens/Host/Host.jsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      { name: i18n._(t`Facts`), link: `${match.url}/facts`, id: 1 },
+      { name: i18n._(t`Groups`), link: `${match.url}/groups`, id: 2 },
+      {
+        name: i18n._(t`Completed Jobs`),
+        link: `${match.url}/completed_jobs`,
+        id: 3,
+      },
+    ];
+
+    const CardHeader = styled(PFCardHeader)`
```
Comment (keithjgrant) [inline awx/ui_next/src/screens/Host/Host.jsx]: This should probably go up near the top of file outside the component definition.

File: awx/ui_next/src/screens/Host/HostList/HostList.jsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+      this.setState({ toggleLoading: false });
+    }
+  }
+
+  async loadHosts() {
+    const { location } = this.props;
+    const { actions: cachedActions } = this.state;
+    const params = parseQueryString(QS_CONFIG, location.search);
+
+    let optionsPromise;
```
Comment (keithjgrant) [inline awx/ui_next/src/screens/Host/HostList/HostList.jsx:123]: A little convoluted here avoiding re-fetching actions... Maybe delegate some of this to a `this.loadActions()` that returns `this.state.actions` if set or fetches if not and extracts them out of the `data` key before returning. Then you could do something like:

`const promises = Promise.all([HostsAPI.read(params), this.loadActions()]);`

**Review by keithjgrant — APPROVED** [review]
Looks good. A couple suggestions to consider noted below.

File: awx/ui_next/src/screens/Host/Host.jsx (hunk 1 above)
Comment (mabashian) [reply awx/ui_next/src/screens/Host/Host.jsx]: There are actually a few spots where this type of thing is defined in render - i'll address all of them

**Comment by mabashian** [comment]
regate

**Comment by mabashian** [comment]
regate

**Pull request merged**